### PR TITLE
Port Records and ModelConfigMeasurement classes

### PR DESCRIPTION
**Pull request by nv-braf**
Ports the Record and ModelConfigMeasurement classes into GAP.

The Record class is used to store individual metrics (like latency or throughput).
The MCM class is used to store all the perf metrics for a model configuration and has methods to compare configurations, change objectives and checkpoint.

### CODE REVIEW

File: genai-perf/tests/test_model_config_measurement.py [hunk 1]
```diff
@@ -0,0 +1,207 @@
+# Copyright 2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
```
Comment (debermudez) [inline genai-perf/tests/test_model_config_measurement.py:1]: With so many additions, there is not a lot of unit testing.
Do you exercise every type of record?

Comment (nv-braf) [reply genai-perf/tests/test_model_config_measurement.py:1]: The record class has never had direct unit testing (it relies on indirect testing as it is used in a variety of unit testbenches). The MCM unit testing covers almost all of the base classes methods. And, the individual records will be exercised in unit tests once we port/create the code that handles the conversion of PA results to RCM/MCM.

Comment (nv-braf) [reply genai-perf/tests/test_model_config_measurement.py:1]: Good catch. There actually is some record type unit testing that I missed porting over. I'll add this on Monday.

Comment (nv-braf) [reply genai-perf/tests/test_model_config_measurement.py:1]: I've added the missing testing (and it also uncovered that I missed a file!). Please re-review when you get a chance.

**Review by debermudez — APPROVED** [review]

**Pull request merged**